### DISCUSSION TITLE
Record and playback MIDI using MIDI commands

### DISCUSSION THREAD

**Discussion by newVideophile** [GrandOrgue/grandorgue#1328] [Help] [answered]
Is there a way to record and playback MIDI without using G.O. UI but MIDI commands only? 

I am asking because I use a headless Raspberry Pi, and I control G.O. using MIDI commands coming from an Arduino Mega with an alphanumeric display and a few buttons used as a simplified UI (see [https://github.com/GrandOrgue/grandorgue/discussions/875#discussioncomment-3300854](url))

I can start to record MIDI using the Recorder/player panel, because there is a "Start" button that can be controlled by a MIDI command. The same applies to stop the recording. Then I get in trouble: G.O. asks me the file name to save the recording. And I can't answer through MIDI commands. Then, if I want to play back, I need to go to the menu "Load MIDI" and select the file. Again, this can't be done using MIDI commands. Finally, I can start to play the file using the "Play" button, which can be MIDI controlled. 

On a side note, I find that the "Recorder / Player" panel behavior is not consistent because some actions can be achieved using MIDI controls, other actions can't. So I don't understand the purpose of allowing to start a recording with a MIDI command, but not to stop it.

**Comment by oleg68** [comment]
For now, there is no way to control choosing a file from midi.

We allow to control every button with midi regardless whether pressing this button is sufficient for some `long` use case or not and we are not going to change it.

Automatic generating a file name based on the timestamp would be useful for controlling buttons with midi, but GO does not have this feature now.

**Comment by willeke1234** [comment, marked as answer]
The MIDI Recorder buttons on the Recorder/Player panel are "REC", "STOP" and "REC File". Which one do you use as "Start" button? If you use the "REC" button then the file name is automatically generated based on the timestamp.

Do you want to choose a file to play or do you want to automatically load the recorded file?

**Reply by newVideophile** [reply]
Thank you, this is exactly what I need !

@oleg68 : so GrandOrgue does have timestamp based filenames.
